### PR TITLE
Update py311

### LINKED ISSUE #78: Support for latest Python and core libraries
Need to make sure that we support the latest Python (3.11) and recent versions of core libraries. We need to remove the version restrictions of some libraries and make sure everything works fine:
- scipy<=1.9.3
- numpy<=1.24.3

TODO:
- [ ] Upgrade the `requirements.txt` file to remove restrictions
- [ ] Upgrade Dockerfile to use Python 3.11
- [ ] Make GitHub CI run tests on multiple Python versions
- [ ] Test and fix any bugs found, if any

### PR DESCRIPTION
**Pull request by EdenWuyifan**
fix #78 
TODO:
- [x] Upgrade the `requirements.txt` file to remove restrictions
- [x] Upgrade Dockerfile to use Python 3.11
- [x] Make GitHub CI run tests on multiple Python versions
- [ ] Test and fix any bugs found, if any
  - [x] Time-series forecasting task dependencies issue and NHITS/NBEATS not working anymore. (Fix in 79a9110242dac522dca180e75a96e64e814517b1)

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ["3.8"]
+        python-version: ["3.8", "3.11"]
```
Comment (EdenWuyifan) [inline .github/workflows/build.yml:11]: @roquelopez Why do we need 3.8 here? Are there envs depends on py3.8?

Comment (aecio) [reply .github/workflows/build.yml:11]: The CI will run the tests against all versions listed here. So, ideally, we should include all Python versions that we want to make sure that `alpha-automl` is compatible with. I think we should include all major versions from 3.8 through 3.11.

File: tests/test_datetime_encoder.py [hunk 2]
```diff
@@ -18,14 +22,18 @@ def test_cyclical_feature(self):
...
     def test_dummy_encoder(self):
         encoder = DummyEncoder()
         encoder.fit(self.df)
         np_array = encoder.transform(self.df)
-        print(np_array)
-        assert np_array.shape == (len(self.df), 4 * len(self.df) + 4)
+        if int(pd.__version__[0]) < 2:
+            assert np_array.shape == (len(self.df), 4 * len(self.df) + 4)
+        else:
+            assert np_array.shape == (len(self.df), 4 * len(self.df))
```
Comment (aecio) [inline tests/test_datetime_encoder.py:39]: Why is there a difference here? Is it worth documenting the difference here?

Comment (roquelopez) [reply tests/test_datetime_encoder.py:39]: Same question as Aecio.

**Review by roquelopez — APPROVED** [review]

**Pull request merged**